tests/http_server_test.cc

```diff
@@ -469,9 +468,8 @@ TEST(
   const size_t CLIENT_REQUEST_SIZE = 3;
   clientLogicFunc(CLIENT_REQUEST_SIZE, server_address, 1, 6);
 
+  std::this_thread::sleep_for(std::chrono::seconds(2));
```

Comment:
Instead of sleeping a fixed amount of time, can you wait (with timeout) on a semaphore?  sleeping like this will slow down the unit tests (slightly, but it can add up) and make them brittle.  If ran on a machine that is oversubscribed for CPU resources (eg, load avg > # logical CPU cores), then this test can fail due to CPU starvation.

Reply:
I'll try to dig more deeply into the logic behind this test when I get some time but I do agree with @dennisjenkins75 

When tests are running asychronously, instead of waiting an arbitrary amount of time (which will slow down unit-tests like Dennis said), I prefer using a `std::mutex` with a `std::condition_variable` to wait on for a period of time (timeout) before making the test fail.

For example, maybe could you just lock the `activeConnections` and add

```
template<typename Duration>
bool waitForActiveConnections(size_t count, Duration timeout);
```

function and then assert on that:

```
ASSERT_TRUE(handler->waitActiveConnections(CLIENT_REQUEST_SIZE, std::chrono::seconds(2)));
```